### PR TITLE
enhance: support null value in index

### PR DESCRIPTION
**Pull request by smellthemoon**
#31728

### CODE REVIEW

**Comment by smellthemoon** [comment]
rerun ut

File: internal/core/src/index/BitmapIndex.cpp
Comment (tedxu) [inline internal/core/src/index/BitmapIndex.cpp:1]: We still need additional handling on the `In` and `NotIn` primitives. The handling is expected to be the same as `In` and `NotIn` function evaluations. I did a testrun under MySQL:

```
+----------------+--------------------+----------------+--------------------+-------------------+-----------------------+
| NULL in (3, 4) | NULL NOT IN (3, 4) | 3 IN (3, NULL) | 3 NOT IN (3, NULL) | NULL IN (3, NULL) | NULL NOT IN (3, NULL) |
+----------------+--------------------+----------------+--------------------+-------------------+-----------------------+
|           NULL |               NULL |              1 |                  0 |              NULL |                  NULL |
+----------------+--------------------+----------------+--------------------+-------------------+-----------------------+
```

File: internal/core/src/index/ScalarIndexSort.cpp
Comment (tedxu) [inline internal/core/src/index/ScalarIndexSort.cpp:1]: This index may have the same issue with the `BitmapIndex`

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**